packages/ethereum/contracts/HoprChannels.sol

```diff
@@ -523,30 +523,55 @@ contract HoprChannels is IERC777Recipient, ERC1820Implementer {
         }
     }
 
+    /**
+     * Uses the response to recompute the challenge. This is done
+     * by multiplying the base point of the curve with the given response.
+     * Due to the lack of embedded ECMUL functionality in the current
+     * version of the EVM, this is done by misusing the `ecrecover` 
+     * functionality. `ecrecover` performs the point multiplication and 
+     * converts the output to an Ethereum address (sliced hash of the product
+     * of base point and scalar).
+     * @param response response that is used to recompute the challenge
+     */
+    function computeChallenge(bytes32 response) public pure returns (address)  {
+        // Field order of the base field
+        uint256 FIELD_ORDER = 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEBAAEDCE6AF48A03BBFD25E8CD0364141;
+
+        // x-coordinate of the base point
+        uint256 gx = 0x79BE667EF9DCBBAC55A06295CE870B07029BFCDB2DCE28D959F2815B16F81798;
+        // y-coordinate of base-point is even, so v is 27
+        uint8 gv = 27;
+
+        address signer = ecrecover(0, gv, bytes32(gx), bytes32(mulmod(uint256(response), gx, FIELD_ORDER)));
+
+        return signer;
+    }
     /**
      * @dev Encode ticket data
      * @return bytes
      */
     function _getEncodedTicket(
         address recipient,
-        uint256 ticketEpoch,
-        bytes32 proofOfRelaySecret,
-        uint256 channelEpoch,
+        uint256 recipientCounter,
+        bytes32 porSecret,
+        uint256 channelIteration,
```

Comment:
if you are renaming this, please align the design document to use the same terminology.

Reply:
changed